urbanairship/urbanairship-cordova
# [MOBILE-881] Adds EU cloud site support to Cordova

### What do these changes do?
This adds EU cloud support to Cordova.

### Why are these changes necessary?
Customers using the EU cloud site need the Cordova SDK to support it. These changes allow for that.

### How did you verify these changes?
I edited the individual plugins for each platform in my own Cordova app and debugged to make sure that the site given (either by the config.xml file or through takeoff) is correctly parsed and set in the individual iOS and Android SDKs.

Line 325 in PluginManager.java and line 134 in UACordovaPluginManager.m are the places where the site is set in the respective standalone SDKs. I made sure that the correct values are passed to these.

To test:

Test 1:
1. Clone this branch and add it as a plugin to your Cordova project.
2. In the config.xml file, add a preference element with the name "com.urbanairship.site" and set the value to "EU".

Example: 
`<preference name="com.urbanairship.site" value="EU" />`

3. Add a breakpoint a couple of lines before the lines I mention above (Lines 325 and 134).
4. Test to make sure `setSite` on Android receives `AirshipConfigOptions.SITE_EU;` and `airshipConfig.site` is equal to the enum value `UACloudSiteEU `.

Test 2:
1. In the config.xml file, comment out the preference element (as well as any app key elements) you added in the last test.
2. Call `UAirship.takeoff()` passing an object as an argument with your app key and secret as well as key called 'site'. 

Here's an example:

```
         {
            production: {
              appKey: "<APP_KEY>",
              appSecret: "<APP_SECRET>"
            },
            development: {
              appKey: "Some app key",
              appSecret: "Some app secret"
            },
            site: "EU"
         }
```

3. Repeat steps 3 and 4 from Test 1.
